libafl/src/inputs/multi.rs

```diff
@@ -53,24 +63,46 @@ impl<I> MultipartInput<I> {
 
     /// Get the individual parts of this input.
     #[must_use]
+    #[inline]
     pub fn parts(&self) -> &[I] {
         &self.parts
     }
 
+    /// Get the individual parts of this input.
+    #[must_use]
+    #[inline]
+    pub fn parts_mut(&mut self) -> &mut [I] {
+        &mut self.parts
+    }
+
+    /// Get a specific part of this input by index.
+    #[must_use]
+    #[inline]
+    pub fn part_by_idx(&self, idx: usize) -> Option<&I> {
+        self.parts.get(idx)
+    }
+
+    /// Get a specific part of this input by index.
+    #[must_use]
+    #[inline]
+    pub fn part_by_idx_mut(&mut self, idx: usize) -> Option<&mut I> {
+        self.parts.get_mut(idx)
+    }
+
     /// Access multiple parts mutably.
     ///
     /// ## Panics
     ///
     /// Panics if idxs is not sorted, has duplicate elements, or any entry is out of bounds.
     #[must_use]
-    pub fn parts_mut<const N: usize>(&mut self, mut idxs: [usize; N]) -> [&mut I; N] {
+    pub fn parts_by_idxs_mut<const C: usize>(&mut self, mut idxs: [usize; C]) -> [&mut I; C] {
```

Comment:
I don't think `by_idxs` is a good name.
`parts_at_offsets`? `parts_at_indexes`? Or just keep `parts`?

Reply:
I guess we do use `idx` across the codebase, so something with idx is fine? Rust uses `index` though